webdriver/tests/bidi/browsing_context/context_created/original_opener.py

```diff
@@ -0,0 +1,58 @@
+import pytest
+from webdriver.bidi.modules.script import ContextTarget
+
+from .. import assert_browsing_context
+
+pytestmark = pytest.mark.asyncio
+
+CONTEXT_CREATED_EVENT = "browsingContext.contextCreated"
+
+
+@pytest.mark.parametrize("type_hint", ["tab", "window"])
+async def test_original_opener_null(bidi_session, wait_for_event, wait_for_future_safe, subscribe_events, type_hint):
+
+    await subscribe_events([CONTEXT_CREATED_EVENT])
+    on_entry = wait_for_event(CONTEXT_CREATED_EVENT)
+
+    top_level_context = await bidi_session.browsing_context.create(type_hint=type_hint)
+
+    context_info = await wait_for_future_safe(on_entry)
+
+    assert_browsing_context(
+        context_info,
+        # We use None here as evaluate no always returns value
+        context=top_level_context["context"],
+        children=None,
+        url="about:blank",
+        parent=None,
+        user_context="default",
+        originalOpener=None
+    )
+
+
+@pytest.mark.parametrize("type_hint", ["tab", "window"])
+@pytest.mark.parametrize("window_features", ["", "popup", "noopener", "noreferrer"])
+async def test_original_opener_present(bidi_session, wait_for_event, wait_for_future_safe, subscribe_events, type_hint, window_features):
+
+    top_level_context = await bidi_session.browsing_context.create(type_hint=type_hint)
+
+    await subscribe_events([CONTEXT_CREATED_EVENT])
+    on_entry = wait_for_event(CONTEXT_CREATED_EVENT)
+
+    await bidi_session.script.evaluate(
+        expression=f"""window.open("", undefined, "{window_features}");""",
+        target=ContextTarget(top_level_context["context"]),
+        await_promise=False)
+
+    context_info = await wait_for_future_safe(on_entry)
+
+    assert_browsing_context(
+        context_info,
+        # We use None here as evaluate no always returns value
```

Comment:
Why does it not always return a valid context? Is that maybe a bug?

Reply:
`window.open` can return `null` and it depends on the provided params.
![image](https://github.com/web-platform-tests/wpt/assets/34244704/8d9d1751-6658-4cd5-b868-f06f3229b257)

Reply:
Then we should add another argument to the parameterize section which specifies if the context is valid or not.

Reply:
> Then we should add another argument to the parameterize section which specifies if the context is valid or not.

I don't understand what the request here is.

Reply:
Specify which feature expects a valid context to be returned and which feature just returns `None`. Then we can also do the appropriate checks for context validity.